src/tools/cleaner.js

```diff
@@ -43,11 +43,35 @@ const recurseReplaceBank = (obj, bank = {}) => {
     if (typeof out !== 'string') {
       return out;
     }
+
     Object.keys(bank).forEach(key => {
       // Escape characters (ex. {{foo}} => \\{\\{foo\\}\\} )
-      const s = String(key).replace(/[-[\]/{}()\\*+?.^$|]/g, '\\$&');
-      const re = new RegExp(s, 'g');
-      out = out.replace(re, bank[key]);
+      const escapedKey = String(key).replace(/[-[\]/{}()\\*+?.^$|]/g, '\\$&');
+      const matchesKey = new RegExp(escapedKey, 'g');
+
+      if (!matchesKey.test(out)) {
+        return;
+      }
+
+      const valueParts = out.split(/({{.*?}})/).filter(Boolean);
+      const replacementValue = bank[key];
+      const isPartOfString = valueParts.length > 1;
+
+      if (isPartOfString) {
+        if (_.isArray(replacementValue) || _.isPlainObject(replacementValue)) {
+          throw new TypeError(
+            [
+              'Cannot interpolate objects or arrays into a string.',
+              `You've sent an ${getObjectType(replacementValue)},`,
+              `which will get coerced to ${replacementValue}`
+            ].join(' ')
+          );
+        }
+      }
+
+      out = isPartOfString
+        ? valueParts.join('').replace(matchesKey, replacementValue)
```

Comment:
Looks like `valueParts.join('')` always equals to `out` here. So can we simplify this line to the following?

```suggestion
        ? out.replace(matchesKey, replacementValue)
```

Reply:
Doh. Nice catch @eliangcs